datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireItemAdapterTest.kt

```diff
@@ -348,4 +354,52 @@ class QuestionnaireItemAdapterTest {
       )
       .isFalse()
   }
+
+  @Test
+  fun onCreateViewHolder_customViewType_shouldReturnCorrectCustomViewHolder() {
+    val viewFactoryMatchers = getQuestionnaireItemViewHolderFactoryMatchers()
+    val questionnaireItemAdapter = QuestionnaireItemAdapter(viewFactoryMatchers)
+    val actualQuestionnaireItemViewHolder =
+      questionnaireItemAdapter.onCreateViewHolder(
+        mock(),
+        QuestionnaireItemViewHolderType.values().size
+      )
+    assertNotNull(actualQuestionnaireItemViewHolder)
+    assertEquals(viewFactoryMatchers[0].factory.create(mock()), actualQuestionnaireItemViewHolder)
+  }
+
+  @Test(expected = QuestionnaireItemAdapter.InvalidQuestionnaireWidgetTypeException::class)
+  fun onCreateViewHolder_customViewType_shouldThrowExceptionForInvalidWidgetType() {
+    val questionnaireItemAdapter =
+      QuestionnaireItemAdapter(getQuestionnaireItemViewHolderFactoryMatchers())
+    questionnaireItemAdapter.onCreateViewHolder(
+      mock(),
+      QuestionnaireItemViewHolderType.values().size + 1
```

Comment:
this fails because of the size of getQuestionnaireItemViewHolderFactoryMatchers() right? so i think you should extract an variable and do this:

```suggestion
      QuestionnaireItemViewHolderType.values().size + customFactoryMatchers.sizes
```

Reply:
done